### PR TITLE
feat(build): Docker build improvements

### PR DESCRIPTION
**Pull request by khvn26**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Add PR comments with Docker image build status.
- Run E2E for ARM images.
- Remove `pip` from runtime layers to reduce attack surface (and have nicer Docker Scout reports).
- Bump `requests` to a non-yanked version.
- Remove `pip-tools` (and pip as transient dependency) from development dependencies.

## How did you test this code?

The builds are tested in CI.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -106,8 +110,11 @@ RUN --mount=type=secret,id=github_private_cloud_token \
 # * api-runtime
 FROM python-slim as api-runtime
 
+# Uninstall pip to reduce CVE-2018-20225 noise
+# and install binary dependencies required for ARM64
 ARG TARGETARCH
-RUN if [ "${TARGETARCH}" != "amd64" ]; then \
+RUN pip uninstall -y pip && \
+  if [ "${TARGETARCH}" != "amd64" ]; then \
   apt-get update && apt-get install -y libpq-dev && rm -rf /var/lib/apt/lists/*; fi;
```
Comment (dabeeeenster) [inline Dockerfile]: I wonder if we still need to install these for arm?

Comment (khvn26) [reply Dockerfile]: We might not https://github.com/psycopg/psycopg2/pull/1545

Comment (khvn26) [reply Dockerfile]: E2E passes on ARM without it 👍

File: frontend/Makefile [hunk 2]
```diff
@@ -30,4 +30,4 @@ serve:
 test:
 	docker compose run frontend \
 		npx cross-env E2E_CONCURRENCY=${E2E_CONCURRENCY} npm run test -- $(opts) \
-		|| docker compose logs flagsmith-api
+		|| (docker compose logs flagsmith-api; exit 1)
```
Comment (gagantrivedi) [inline frontend/Makefile:33]: Why did we add this? How was this working before?

**Review by gagantrivedi — APPROVED** [review]
Approved with a question

File: frontend/Makefile (hunk 2 above)
Comment (khvn26) [reply frontend/Makefile:33]: `docker compose logs flagsmith-api` exited 0, making failed E2E jobs green.

**Pull request merged**